checker/tests/nullness/Issue577.java

```diff
@@ -0,0 +1,82 @@
+// Test case for issue #577 with expected errors:
+// Also see test case in framework/tests/all-systems/Issue577.java
+// https://github.com/typetools/checker-framework/issues/577
+
+import org.checkerframework.checker.nullness.qual.*;
+
+class Banana<T extends Number> extends Apple<int[]> {
+    @Override
+    void fooOuter(int[] array) {}
+    class InnerBanana extends InnerApple<long[]> {
+        @Override
+        //:: error: (override.param.invalid)
+        <F2 extends Object> void foo(int[] array, long[] array2, F2 param3) {
+        }
+    }
+}
+
+class Apple<T> {
+    void fooOuter(T param) {}
+    class InnerApple<E> {
+        <F> void foo(T param, E param2, F param3) {
+        }
+    }
+}
+
+class Pineapple<E extends Object> extends Apple<E> {
+   @Override
+   void fooOuter(E array) {}
+   class InnerPineapple extends InnerApple<@Nullable String> {
+        @Override
+        //:: error: (override.param.invalid)
+        <F3> void foo(E array, String array2, F3 param3) {
+        }
+    }
+}
+
+class IntersectionAsMemberOf {
+    interface MyGenericInterface<F> {
+        F getF();
+    }
+
+    <T extends Object & MyGenericInterface<@NonNull String>> void foo(T param) {
+      //TODO: this error is due to a bug in asSuper
```

Comment:
I'm a bit weary of TODOs in test files. Would it make sense to extract these into separate test files that are skipped? That is a bit more explicit about work still being required.